### PR TITLE
refactor: getting taquito oxford ready

### PR DESCRIPTION
**Pull request by hui-an-yang**
prepping this branch oxford ready

Thank you for your contribution to Taquito.

Before submitting this PR, please make sure:

- [x] Your code builds cleanly without any errors or warnings
- [x] You have run the linter against the changes
- [ ] You have added unit tests (if relevant/appropriate)
- [ ] You have added integration tests (if relevant/appropriate)
- [ ] All public methods or types have TypeDoc coverage with a complete description, and ideally an @example
- [ ] You have added or updated corresponding documentation
- [x] If relevant, you have written a first draft summary describing the change for inclusion in Release Notes. 

In this PR, please also make sure: 

- [ ] You have linked this PR to the issue by putting `closes #TICKETNUMBER` in the description box (when applicable)
- [x] You have added a concise description on your changes
## Release Note Draft Snippet

__If relevant, please write a summary of your change that will be suitable for
inclusion in the Release Notes for the next Taquito release.__

### CODE REVIEW

File: docs/wallet_API.md [hunk 1]
```diff
@@ -161,7 +161,7 @@ const wallet = new TempleWallet('MyAwesomeDapp');
 The class constructor takes one parameter, the name of your dapp (this will be used later in the transaction confirmation pop-up). After the instantiation, we can connect the wallet by calling the `connect` method:
 
 ```js
-await wallet.connect('mainnet' | 'mumbainet'  | 'nairobinet' | 'ghostnet' | 'mondaynet' | 'sandbox');
+await wallet.connect('mainnet' | 'oxfordnet'  | 'oxfordnet' | 'ghostnet' | 'mondaynet' | 'sandbox');
```
Comment (AlirezaHaghshenas) [inline docs/wallet_API.md:164]: Two `oxfordnet`s

File: docs/maps_bigmaps.md [hunk 2]
```diff
@@ -28,7 +28,7 @@ Michelson offers two variants of `Maps` that are semantically the same but have
 
 This example builds on the Ligo Lang Taco Shop learning resources.
 
-The storage of the contract used in the following example is a map where a key is a natural number (a `nat`), and a value is a pair composed of two values representing the quantity of stock and `tez` tokens, respectively. The contract's source code is available [here](https://ligolang.org/docs/tutorials/get-started/tezos-taco-shop-smart-contract#making-sure-we-get-paid-for-our-tacos). In the example, the contract is originated with initial values using the `MichelsonMap` class' `set` method.
+The storage of the contract used in the following example is a map where a key is a natural number (a `nat`), and a value is a pair composed of two values representing the quantity of stock and `tez` tokens, respectively. The contract code is available [here](https://ligolang.org/docs/tutorials/get-started/tezos-taco-shop-smart-contract#making-sure-we-get-paid-for-our-tacos). In the example, the contract is originated with initial values using the `MichelsonMap` class' `set` method.
```
Comment (dsawali) [inline docs/maps_bigmaps.md:31]: slight nitpick, but I'd phrase the first sentence this way instead, "... example is a map where the key is a naturam number (`nat`), and the value is a pair composed of...

File: docs/smart_rollups.md [hunk 3]
```diff
@@ -7,7 +7,7 @@ author: Davis Sawali
 
 Rollups are a permissionless scaling implementation for the Tezos blockchain. The idea is that anyone can originate and operate one or more rollups, increasing the overall throughput of the Tezos blockchain.
 
-In Taquito, we have implemented some of the operations included in Mumbai protocol update in regards to smart rollups. In this document, we will go through the operations we support. We also won't go too detailed on how rollups work behind the scenes, if you'd like to understand the feature a bit deeper, you can refer to [this document](https://tezos.gitlab.io/mumbai/smart_rollups.html).
+In Taquito, we have implemented some of the operations included in Mumbai protocol update in regards to smart rollups. In this document, we will go through the operations we support. We also won't go too detailed on how rollups work behind the scenes, if you'd like to understand the feature a bit deeper, you can refer to [this document](https://tezos.gitlab.io/oxford/smart_rollups.html).
```
Comment (dsawali) [inline docs/smart_rollups.md:10]: paragraph still mentions Mumbai, we should probably rephrase it in a way where we clearly explain that the smart rollup changes were initially rolled out in the Mumbai update but the feature still lives in current protocols

File: docs/smart_rollups.md [hunk 4]
```diff
@@ -27,10 +27,11 @@ await op.confirmation();
 
 ##### please note
 - `pvmKind` at this time the only PVM supported is `wasm_2_0_0`
-- `kernel` is passed as a hexadecimal string examples can be found at [this tezos docs endpoint](https://tezos.gitlab.io/mumbai/smart_rollups.html?highlight=smart%20rollup%20originate#id1)
+- `kernel` is passed as a hexadecimal string examples can be found at [this tezos docs endpoint](https://tezos.gitlab.io/oxford/smart_rollups.html?highlight=smart%20rollup%20originate#id1)
 - `parametersType` is a MichelsonV1Expression to define the type.
+-  **Please note** `originationProof` is needed origination in `batch` which can be obtained with the `rpc.getOriginationProof({kernel, kind: PVMKind})`
```
Comment (dsawali) [inline docs/smart_rollups.md:32]: I thought this param was removed in Oxford?

File: integration-tests/contract-failing-noop.spec.ts [hunk 5]
```diff
@@ -1,34 +1,30 @@
...
-import { CONFIGS, defaultSecretKey, isSandbox } from "./config";
-import { OpKind, Protocols, TezosToolkit } from "@taquito/taquito";
+import { CONFIGS, defaultSecretKey } from "./config";
+import { OpKind, TezosToolkit } from "@taquito/taquito";
 import { verifySignature } from "@taquito/utils";
 
-CONFIGS().forEach(({ rpc, setup, protocol}) => {
-
+CONFIGS().forEach(({ setup }) => {
+  let rpc = 'https://mainnet-archive.ecadinfra.com/'
```
Comment (dsawali) [inline integration-tests/contract-failing-noop.spec.ts:7]: Could we add this into the new config that Arvid help build? Seems like we can probably move away from using direct string variables for urls, unless there's something preventing us from doing so.

File: packages/taquito-local-forging/src/taquito-local-forging.ts [hunk 6]
```diff
@@ -22,7 +22,7 @@ export * from './interface';
 export { VERSION } from './version';
 export { ProtocolsHash } from './protocols';
 
-const PROTOCOL_CURRENT = ProtocolsHash.PtMumbai2;
+const PROTOCOL_CURRENT = ProtocolsHash.Proxford;
```
Comment (dsawali) [inline packages/taquito-local-forging/src/taquito-local-forging.ts:25]: correct me if I'm wrong, but isn't the shortened protocol hash `ProxfordS` instead?

Comment (hui-an-yang) [reply packages/taquito-local-forging/src/taquito-local-forging.ts:25]: it's defined [here](https://github.com/ecadlabs/taquito/blob/oxford/packages/taquito-local-forging/src/protocols.ts#L18) 
image

File: integration-tests/contract-failing-noop.spec.ts (hunk 5 above)
Comment (hui-an-yang) [reply integration-tests/contract-failing-noop.spec.ts:7]: What arvid have refactored is for testnets, this is the set up. Think we don't mind querying rpc for data from mainnet but we don't want to run test with configSigner or secretKey on mainnet.
image

File: docs/smart_rollups.md (hunk 4 above)
Comment (hui-an-yang) [reply docs/smart_rollups.md:32]: Thanks for caching, will remove. It's removed in [oxford branch]( https://github.com/ecadlabs/taquito/pull/2661/files#diff-d8805c22ac4e0fde66a9c098bdd50de333641a472ddae1d37b0a74f042fb2b67L32) seems overwriting by auto-merging. 
image

File: docs/smart_rollups.md (hunk 3 above)
Comment (hui-an-yang) [reply docs/smart_rollups.md:10]: great idea, will rephrase the sentence as 

`In Taquito, we have implemented some of the smart rollups operations which are first introduced in Mumbai protocol.`

File: docs/maps_bigmaps.md (hunk 2 above)
Comment (hui-an-yang) [reply docs/maps_bigmaps.md:31]: thanks for the suggestions, will update

File: docs/wallet_API.md (hunk 1 above)
Comment (hui-an-yang) [reply docs/wallet_API.md:164]: good catch, thank you will update

**Review by AlirezaHaghshenas — APPROVED** [review]

File: packages/taquito-local-forging/src/taquito-local-forging.ts (hunk 6 above)
Comment (hui-an-yang) [reply packages/taquito-local-forging/src/taquito-local-forging.ts:25]: Counted some 6 some 7. But you're right the recent ones are all 7 char, will update

**Review by dsawali — APPROVED** [review]

**Pull request merged**